pgmpy/prediction/NaiveBackdoorRegressor.py

```diff
@@ -0,0 +1,284 @@
+            return feature_df
+
+        # Cases 1 & 2: Standard semantic mapping from named columns
+        missing_columns = set(required_features) - set(X_df.columns)
+        if missing_columns:
+            raise ValueError(
+                f"Missing required columns in input data: {list(missing_columns)}. "
+                f"Required columns based on DAG roles: {required_features}."
+            )
+
+        return X_df[required_features]
+
+    def fit(self, X, y, sample_weight: Optional[np.ndarray] = None, feature_names=None):
+        """
+        Fit the Naive Backdoor Regressor.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features) or pandas DataFrame
+            Training data. If DataFrame, must include columns for exposure,
+            adjustment, and pretreatment variables as defined in causal graph roles.
+            If array, `feature_names` must be provided unless used within a
+            generic sklearn context (like compatibility tests).
+        y : array-like of shape (n_samples,)
+            Target values (outcome variable).
+        sample_weight : array-like of shape (n_samples,), optional
+            Sample weights.
+        feature_names : list, optional
+            Feature names when X is an array. Strongly recommended for array inputs
+            to ensure correct semantic mapping.
+
+        Returns
+        -------
+        self
+            Returns self for method chaining.
+        """
+        X_arr, y_arr = check_X_y(
+            X, y, accept_sparse=False, ensure_2d=True, dtype="numeric"
+        )
+
+        # Extract and validate causal graph roles and define required features
+        exposure_var, outcome_var, adjustment_vars, pretreatment_vars = (
+            self._validate_dag_and_extract_roles()
+        )
+        self.exposure_var_ = exposure_var
+        self.outcome_var_ = outcome_var
+        self.adjustment_vars_ = adjustment_vars
+        self.pretreatment_vars_ = pretreatment_vars
+        self.feature_columns_ = self._extract_feature_columns()
+
+        # Prepare the feature DataFrame using the centralized, safe logic
+        X_features = self._prepare_feature_df(X, feature_names)
+
+        # Set sklearn attributes. n_features_in_ should reflect the original input shape.
+        self.n_features_in_ = X_arr.shape[1]
+        self.feature_names_in_ = np.array(X_features.columns, dtype=object)
+
+        # Initialize base estimator
+        self.estimator_ = (
+            LinearRegression()
+            if self.base_estimator is None
+            else clone(self.base_estimator)
+        )
+
+        # Fit the base estimator on the prepared features
+        fit_params = {}
+        if sample_weight is not None:
+            fit_params["sample_weight"] = sample_weight
+
+        self.estimator_.fit(X_features, y_arr, **fit_params)
+
+        return self
+
+    def predict(self, X, feature_names=None):
```

Comment:
The `predict` method also requires a validation of `X` using the `validate_data` method.